client/components/auth/router.decorator.js

```diff
@@ -12,20 +12,50 @@ export function routerDecorator($rootScope, $location, Auth) {
     }
 
     if (typeof next.authenticate === 'string') {
-      Auth.hasRole(next.authenticate)
-        .then(has => {
-          if (has) {
-            return;
-          }
+      if (next.authenticate === 'student'){
+        return Auth.isStudent()
+          .then(has => {
+            if (has) {
+              return;
+            }
 
-          event.preventDefault();
-          return Auth.isLoggedIn()
-            .then(is => {
-              $location.path(is ? '/' : '/login');
-            });
-        });
+            event.preventDefault();
+            return Auth.isLoggedIn()
+              .then(is => {
+                $location.path(is ? '/' : '/login');
```

Comment:
Those were the names already used, I just duplicated them for consistency. Do you have a better name in mind?